java-checks/src/main/java/org/sonar/java/checks/SwitchCasesShouldBeCommaSeparatedCheck.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.sonar.check.Rule;
+import org.sonar.java.JavaVersionAwareVisitor;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.JavaVersion;
+import org.sonar.plugins.java.api.tree.CaseGroupTree;
+import org.sonar.plugins.java.api.tree.CaseLabelTree;
+import org.sonar.plugins.java.api.tree.SwitchTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+@Rule(key = "S6208")
+public class SwitchCasesShouldBeCommaSeparatedCheck extends IssuableSubscriptionVisitor implements JavaVersionAwareVisitor {
+  private static final String MESSAGE = "Merge the previous cases into this one using comma-separated label.";
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return Arrays.asList(
+      Tree.Kind.SWITCH_EXPRESSION,
+      Tree.Kind.SWITCH_STATEMENT
+    );
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    SwitchTree switchExpression = (SwitchTree) tree;
+    if (usesArrows(switchExpression)) {
```

Comment:
Small comment: If we test `!useColons` instead of `useArrows`, it's almost the same code but when java will introduce "yet another switch expression", this rule will continue to work without effort.